### ISSUE TITLE
poor search performance

### ISSUE THREAD

**Issue by invakid404** [WDaan/VueTorrent#540]
I have ~1400 torrents in my QBit, and searching through them takes a long time, especially the initial search when I type the first few characters. The worst part is that the entire UI blocks when doing that, resulting in a very bad experience.

Debouncing the search input will definitely make it a little more bearable, but further benchmarking is necessary to find the actual bottleneck.

**Comment by WDaan** [comment]
Thanks for the PR! Yeah, the quick search is a bit better but searching in general is still slow. Especially when the number of torrents increases.

The UI in general is also noticably slower when you have lots of active torrents.

I'm not so sure yet how to adress either of those issues yet 🙁

**Comment by pinkairfreshener** [comment]
> I have ~1400 torrents in my QBit, and searching through them takes a long time, especially the initial search when I type the first few characters. The worst part is that the entire UI blocks when doing that, resulting in a very bad experience.
> 
> Debouncing the search input will definitely make it a little more bearable, but further benchmarking is necessary to find the actual bottleneck.

I’m literally having the exact same issue and I have about ~1500 torrents.

**Comment by pinkairfreshener** [comment]
I’m on 1.8.0 which is what is bundled with hotio qb but i will try vue3 now, thanks :)

**Comment by pinkairfreshener** [comment]
> @pinkairfreshener Have you tried the Vue3 version yet?

Wait, 1.8.0 is the latest version, what did you mean by vue3?

**Comment by pinkairfreshener** [comment]
Thanks I have pulled the nightly release branch to ~/vuetorrent_nightly. Inside there is now a folder called VueTorrent. Which exact folder do I point QB to? Is it "~/vuetorrent_nightly" OR "~/vuetorrent_nightly/VueTorrent"?

Thanks

**Comment by Larsluph** [comment]
you should have only a public folder

**Comment by pinkairfreshener** [comment]
Forgive me but I’m struggling to install the nightly release - are there any instructions I could find to help with this? Thanks

**Comment by Larsluph** [comment]
Everything's in the readme 😄

**Comment by pinkairfreshener** [comment]
git clone --single-branch --branch nightly-release https://github.com/WDaan/VueTorrent.git

Then when I try npm install, I get an error saying no package.json file found

**Comment by Larsluph** [comment]
yeah because it's already built, you should find a public folder and that's it.

**Comment by pinkairfreshener** [comment]
oh 🙊😂 thanks

**Comment by Larsluph** [comment]
Is it outdated?
